pycbc/events/coinc.py

```diff
@@ -370,7 +370,105 @@ def cluster_coincs(stat, time1, time2, timeslide_id, slide, window, argmax=numpy
 
     span = (time.max() - time.min()) + window * 10
     time = time + span * tslide
+    cidx = cluster_given_mean_times(stat, time, window, argmax)
+    return cidx
+
+def cluster_coincs_multiifo(stat, time_coincs, timeslide_id, slide, window, argmax=numpy.argmax):
+    """Cluster coincident events for each timeslide separately, across
+    templates, based on the ranking statistic
+
+    Parameters
+    ----------
+    stat: numpy.ndarray
+        vector of ranking values to maximize
+    time_coincs: tuple of numpy.ndarrays
+        tuple containing time vectors each ifo
+    timeslide_id: numpy.ndarray
+        vector that determines the timeslide offset
+    slide: float
+        length of the timeslides offset interval
+    window: float
+        legth to cluster over
+
+    Returns
+    -------
+    cindex: numpy.ndarray
+        The set of indices corresponding to the surviving coincidences
+    """
+    time_coinc_zip = zip(*time_coincs)
+    if len(time_coinc_zip) == 0:
+        logging.info('No coincident triggers.')
+        return numpy.array([])
+
+    time_avg_num = []
+    #find the mean of each time coincidences, providing that the IFO is on
+    for tc in time_coinc_zip:
+        time_avg_num.append(mean_if_greater_than_zero(tc))
+
+    time_avg, num_ifos = zip(*time_avg_num)
+
+    time_avg = numpy.array(time_avg)
+    num_ifos = numpy.array(num_ifos)
+
+    # shift all but the pivot ifo - leads to a (num_ifos-1) * timeslide_id * slide shift
+    if numpy.isfinite(slide):
+        nifos_minusone = (num_ifos - numpy.ones_like(num_ifos))
+        time_avg = time_avg + (nifos_minusone * timeslide_id * slide)/num_ifos
+
+    tslide = timeslide_id.astype(numpy.float128)
+    time_avg = time_avg.astype(numpy.float128)
+
+    span = (time_avg.max() - time_avg.min()) + window * 10
+    time_avg = time_avg + span * tslide
+    cidx = cluster_given_mean_times(stat, time_avg, window, argmax)
+    return cidx
+
+def mean_if_greater_than_zero(vector):
+    """ Calculate mean of a vector, but ignore it if the value if it is less than zero
+    This is used when the timestamps are marked as -1 when a particular
+    IFO is not included in the particular coincidence.
+
+    Parameters
+    ----------
+    vector: numpy.ndarray
+        vector to be mean averaged
+
+    Returns
+    -------
+    mean: float
+        The mean of the values in the original vector which are
+        greater than zero
+    num_above_zero: int
+        The number of entries in the vector which are above zero
+    """
+    vals_above_zero = [val for val in vector if val > 0]
+    num_above_zero = len(vals_above_zero)
+    mean = sum(vals_above_zero) / num_above_zero
+    return mean, num_above_zero
 
+def cluster_given_mean_times(stat, time, window, argmax=numpy.argmax):
```

Comment:
since this function doesn't rely on the times being mean times of coincs I would just call it cluster_over_time. 
also for the docstring, can just say 'Cluster events given their times and ranking statistic values'